compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -514,6 +514,18 @@ private BType getFiniteTypeMatchWithIntType(BLangLiteral literalExpr, BFiniteTyp
         } else if (literalAssignableToFiniteType(literalExpr, finiteType, TypeTags.BYTE)) {
             setLiteralValueForFiniteType(literalExpr, symTable.byteType, data);
             return symTable.byteType;
+        } else if (literalAssignableToFiniteType(literalExpr, finiteType, TypeTags.FLOAT)) {
+            if (!(literalExpr.value instanceof Long)) {
+                // to prevent eg: 1f|1d|2d f2 = 1; from passing
```

Comment:
`1f|1d|2d f2 = 1` works. Why should this be disallowed anyway?

Reply:
In master since `1f|1d|2d f1 = 2;` is not allowed. Should this be allowed?

Reply:
`1f|1d|2d f1 = 2;` should fail, not `1f|1d|2d f2 = 1`.

Reply:
Not addressed? If the comment is wrong fix it.

Reply:
Also, this is working on master. Which change of yours resulted in requiring this?